core/src/main/scala/cats/data/EitherT.scala

```diff
@@ -120,6 +120,9 @@ final case class EitherT[F[_], A, B](value: F[Either[A, B]]) {
       case r@Right(_) => F.pure(r.leftCast)
     })
 
+  def biSemiflatMap[C, D](fa: A => F[C], fb: B => F[D])(implicit F: Monad[F]): EitherT[F, C, D] =
+    leftSemiflatMap(fa).semiflatMap(fb)
```

Comment:
Maybe it would be better to implement this in a single walk over `F`?

If `F` is e.g. `List`, this would traverse the list twice, and we should probably not do that ;)

Other than that, I think it's worth mentioning this PR to one of the maintainers, since there hasn't been any activity from them on this or the original issue.

Reply:
Fair point @kubukoz , so I would refactor it to be:
```
def biSemiflatMap[C, D](fa: A => F[C], fb: B => F[D])(implicit F: Monad[F]): EitherT[F, C, D] =
    EitherT(F.flatMap(value) {
      case Left(a) => F.map(fa(a)) { c => Left(c) }
      case Right(b) => F.map(fb(b)) { d => Right(d) }
    })
```
In this case it will `flatMap` and `map` it once, instead of `flatMap` it twice.